examples/hosts/app-integration/external-data/src/view/debugView.tsx

```diff
@@ -64,11 +75,28 @@ const ExternalDataView: React.FC<IExternalDataViewProps> = (props: IExternalData
 
                 const responseBody = await response.json() as Record<string, unknown>;
                 const newData = responseBody.taskList as TaskData;
-                if(newData !== undefined && !isEqual(newData,externalData)) {
+                if (newData !== undefined && !isEqual(newData, externalData)) {
                     console.log("APP: External data has changed. Updating local state with:\n", newData)
+                    const updatedExternalData = Object.entries(newData);
+                    let count = 0;
+                    // eslint-disable-next-line unicorn/no-array-for-each
+                    updatedExternalData.forEach(([id, { name, priority }]) => {
+                        const existing = externalData[id] as {
+                            name: string;
+                            priority: number;
+                        };
+                        if (existing === undefined) {
+                            count++;
+                        } else if (existing.name !== name || existing.priority !== priority) {
+                            count++;
+                        }
+                    });
+                    props.setUnsynchronizedChangesCount(count)
                     setExternalData(newData);
+                } else {
+                    props.setUnsynchronizedChangesCount(0);
                 }
-            } catch(error) {
+            } catch (error) {
                 console.error(`APP: An error was encountered while polling external data:\n${error}`);
```

Comment:
If you change this to  `console.error('APP: An error was encountered while polling external data:', error);` you should get a better view of the error in the console log.